src/lib/CMakeLists.txt

```diff
@@ -0,0 +1,231 @@
+check_include_file(sys/resource.h HAVE_SYS_RESOURCE_H)
+check_include_file(sys/stat.h HAVE_SYS_STAT_H)
+check_include_file(sys/types.h HAVE_SYS_TYPES_H)
+check_include_file(unistd.h HAVE_UNISTD_H)
+set(HAVE_ZLIB_H "${ZLIB_FOUND}")
+set(HAVE_BZLIB_H "${BZIP2_FOUND}")
+configure_file(config.h.in config.h)
+
+# check botan's enabled features
+set(CMAKE_REQUIRED_INCLUDES "${BOTAN2_INCLUDE_DIRS}")
+set(_botan_required_features
+    # base
+    BIGINT FFI HEX_CODEC PGP_S2K
+    # AEAD
+    BLOCK_CIPHER AEAD_EAX AEAD_OCB
+    # symmetric ciphers
+    AES BLOWFISH CAMELLIA CAST_128 DES IDEA RIPEMD_160 SM4 TWOFISH
+    # cipher modes
+    MODE_CBC MODE_CFB
+    # RNG
+    AUTO_RNG AUTO_SEEDING_RNG HMAC HMAC_DRBG
+    # hash
+    CRC24 HASH MD5 SHA1 SHA2_32 SHA2_64 SHA3 SM3
+    # public-key core
+    DL_GROUP DL_PUBLIC_KEY_FAMILY ECC_GROUP ECC_PUBLIC_KEY_CRYPTO PUBLIC_KEY_CRYPTO
+    # public-key algs
+    DSA ECDH ECDSA ED25519 ELGAMAL RSA SM2
+    # public-key operations etc
+    EME_PKCS1v15 EMSA_PKCS1 EMSA_RAW KDF_BASE RFC3394_KEYWRAP SP800_56A
+)
+include(CheckCXXSymbolExists)
+foreach(feature ${_botan_required_features})
+  check_cxx_symbol_exists("BOTAN_HAS_${feature}" botan/build.h _botan_has_${feature})
+  if (NOT _botan_has_${feature})
+    message(FATAL_ERROR "A required botan feature is missing: ${feature}")
+  endif()
+endforeach()
+set(CMAKE_REQUIRED_INCLUDES)
+
+add_library(librnp
+  # librepgp
+  ../librepgp/packet-parse.c
+  ../librepgp/packet-print.c
+  ../librepgp/packet-show.c
+  ../librepgp/reader.c
+  ../librepgp/repgp.c
+  ../librepgp/stream-armor.c
+  ../librepgp/stream-common.c
+  ../librepgp/stream-dump.c
+  ../librepgp/stream-key.c
+  ../librepgp/stream-packet.c
+  ../librepgp/stream-parse.c
+  ../librepgp/stream-sig.c
+  ../librepgp/stream-write.c
+  ../librepgp/validate.c
+
+  # librekey
+  ../librekey/key_store_g10.c
+  ../librekey/key_store_kbx.c
+  ../librekey/key_store_pgp.c
+  ../librekey/key_store_ssh.c
+  ../librekey/rnp_key_store.c
+
+  bufgap.c
+  crypto/bn.c
+  crypto/dsa.c
+  crypto/ec.c
+  crypto/ecdh.c
+  crypto/ecdsa.c
+  crypto/eddsa.c
+  crypto/elgamal.c
+  crypto/hash.c
+  crypto/mpi.c
+  crypto/rng.c
+  crypto/rsa.c
+  crypto/s2k.c
+  crypto/sm2.c
+  crypto/symmetric.c
+  crypto.c
+  fingerprint.c
+  generate-key.c
+  key-provider.c
+  list.c
+  misc.c
+  packet-create.c
+  pass-provider.c
+  pem.c
+  pgp-key.c
+  rnp2.c
+  rnp.c
+  writer.c
+)
+
+set_target_properties(librnp
+  PROPERTIES
+    VERSION "${VERSION}"
+    SOVERSION 0
+    PREFIX ""
+)
+
+target_include_directories(librnp
+  PUBLIC
+    "$<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/src/lib>"
+    "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
+    "$<INSTALL_INTERFACE:include>"
+  PRIVATE
+    "${CMAKE_CURRENT_SOURCE_DIR}"
+    "${PROJECT_SOURCE_DIR}/src"
+)
+
+# This has precedence and can cause some confusion when the binary
+# dir one isn't actually being used. To be improved.
+file(REMOVE "${CMAKE_CURRENT_SOURCE_DIR}/config.h")
```

Comment:
@dewyatt Got hit by this line on Mac (or, probably, because of the way I used to build). While it worked on Travis container It does not work if run as `cmake .`.
Do you have any ideas on improvement?

Reply:
I added it at the last minute honestly. Yes, I can add a workaround - one minute.